universe/Planet.cpp

```diff
@@ -876,8 +876,12 @@ const std::string Planet::OrbitalSuffix() const {
     TemporaryPtr<const System> system_ptr = GetSystem(this->SystemID());
     int orbit = system_ptr->OrbitOfPlanet(this->ID());
     if (m_size == SZ_ASTEROIDS) {  // asteroids are named according to orbit position
-        if (orbit < 1) {
-            orbit = system_ptr->NumPlanets();
+        while (!UserStringExists("NEW_ASTEROID_SUFFIX_ORBIT_" + RomanNumber(orbit))) {
+            --orbit;
+            if (orbit < 1) {
+                orbit = system_ptr->NumPlanets();
+                break;
+            }
```

Comment:
Is this insufficient to catch an infinite loop?  Falling back to UserString handling on non-existent lookup entries.